File: Arrays/3289. The Two Sneaky Numbers of Digitville/3289. The Two Sneaky Numbers of Digitville.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getSneakyNumbers = function (nums) {
  let ans = [];
  for (const n of nums) {
    let i = Math.abs(n);

    if (i === 999) i = 0;
    if (nums[i] === 0) nums[i] = 999;

    if (nums[i] < 0) ans.push(i);
    else nums[i] *= -1;
  }
  return ans;
};

// Time: O(n)
// Space: O(1)

File: Arrays/3289. The Two Sneaky Numbers of Digitville/3289. The Two Sneaky Numbers of Digitville.py
class Solution:
    def getSneakyNumbers(self, nums: List[int]) -> List[int]:   
        ans = []
        for n in nums:
            i = abs(n)
            if i == 999:
                i = 0
            
            if nums[i] == 0:
                nums[i] = 999
            
            if nums[i] < 0:
                ans.append(i)
            
            else:
                nums[i] *= -1

        return ans

# Time: O(n)
# Space: O(1)